src/brsTypes/index.ts

```diff
@@ -39,6 +42,13 @@ export * from "./components/RoInt";
 export * from "./components/RoInvalid";
 export * from "./Callable";
 
+export type NodeCtor = () => RoSGNode;
+
+export const BrsBuiltInComponents = new Map<string, NodeCtor>([
```

Comment:
If there's a better place for this (or if it deserves its own file), let me know!

Reply:
Seems fine.

Reply:
In the future we might need to use the factory pattern to create these but for now I also think it is fine

Reply:
Agreed, with the map I was following the pattern of `BrsObjects`: https://github.com/sjbarag/brs/blob/master/src/brsTypes/components/BrsObjects.ts#L20-L39